extensions/VisualEditor/wikia/ApiAddMediaTemporary.php

```diff
@@ -154,6 +156,14 @@ protected function checkPermissions() {
 		}
 	}
 
+	protected function checkPermissionsForJA() {
```

Comment:
I would suggest moving the business logic into checkPermissions (even though it's called in 2 places) rather than create a new checkPermissionsForJA() function.  the test for if ( JA and Upload and Mobile) can all be in checkPermissions().  And then maybe a unit test to ensure that the function works in those environments.   It looks like checkPermissions() is called in two places, and this logic only applies to one of them, but it's a protected function so it can have a parameter true/false for the different behavior, which makes it more explicit.  

Otherwise, having 2 tests before calling check permissions (as Inez suggested) is also fine.  We have a lot more complicated tests elsewhere for various languages + application states.